src/test/java/com/toopher/TestAuthenticationRequest.java

```diff
@@ -0,0 +1,122 @@
+package com.toopher;
+
+import org.json.JSONObject;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.net.URI;
+import java.util.UUID;
+
+import static org.junit.Assert.*;
+
+public class TestAuthenticationRequest {
+    private static final String DEFAULT_BASE_URL = "https://api.toopher.test/v1/";
+
+    private static String id;
+    private static String reason;
+    private static JSONObject jsonResponse = new JSONObject();
+    private static String userName;
+    private static String terminalName;
+    private static String terminalNameExtra;
+
+    @BeforeClass
+    public static void setUp() {
+        id = UUID.randomUUID().toString();
+        reason = "it is a test";
+
+        JSONObject user = new JSONObject();
+        user.put("id", UUID.randomUUID().toString());
+        user.put("name", "userName");
+        userName = user.getString("name");
+
+        JSONObject terminal = new JSONObject();
+        terminal.put("id", UUID.randomUUID().toString());
+        terminal.put("name", "terminalName");
+        terminal.put("name_extra", "terminalNameExtra");
+        terminal.put("user", user);
+        terminalName = terminal.getString("name");
+        terminalNameExtra = terminal.getString("name_extra");
+
+        JSONObject action = new JSONObject();
+        action.put("id", UUID.randomUUID().toString());
+        action.put("name", "log in");
+
+        jsonResponse.put("id", id);
+        jsonResponse.put("pending", true);
+        jsonResponse.put("granted", false);
+        jsonResponse.put("automated", false);
+        jsonResponse.put("reason", reason);
+        jsonResponse.put("reason_code", 200);
+        jsonResponse.put("user", user);
+        jsonResponse.put("terminal", terminal);
+        jsonResponse.put("action", action);
+    }
+
+    @Test
+    public void testGrantWithOtp() throws InterruptedException, RequestError {
+        JSONObject newJsonResponse = new JSONObject(jsonResponse, JSONObject.getNames(jsonResponse));
+        newJsonResponse.remove("pending");
+        newJsonResponse.remove("granted");
+        newJsonResponse.put("pending", false);
+        newJsonResponse.put("granted", true);
```

Comment:
Can we not set these values directly? Seems easier to do an update than a delete and create combo.